extensions/amp-soundcloud/1.0/storybook/Basic.amp.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../../src/preact';
+import {boolean, color, text, withKnobs} from '@storybook/addon-knobs';
+import {rgba2hex} from '../converter';
+import {withAmp} from '@ampproject/storybook-addon';
+
+export default {
+  title: 'amp-soundcloud-1_0',
+  decorators: [withKnobs, withAmp],
+
+  parameters: {
+    extensions: [{name: 'amp-soundcloud', version: '1.0'}],
+    experiments: ['bento'],
+  },
+};
+
+export const TrackId = () => {
+  // Knobs
+  const componentColor = color('Color', 'RGBA(255, 85, 0, 1)');
+  const height = text('Height', '180');
+  const trackid = text('Track ID', '864765493');
+  const visual = boolean('Visual', true);
+
+  // Convert RGBA to HEX (without Alpha Channel)
+  const hex = rgba2hex(componentColor);
+
+  // Render Preact Component
+  return (
+    <amp-soundcloud
+      color={hex}
+      width="100%"
```

Comment:
@AnuragVasanwala I'm getting error in console for this percentage width, `Invalid "width" value:  100% `.

Reply:
I think percentage width should not be used. The element will need a fixed width & height. Right?

Reply:
I'm trying to find out width percentage usage in existing any amp storybook js files but cannot getting any. According to this line https://github.com/rtCamp/amphtml/blob/main/src/layout.js#L399 we have to use `px` or `auto` for width attribute.

Reply:
Both of you are correct.

Here's, my analysis for the [`parseLength`](https://github.com/rtCamp/amphtml/blob/b2cd4622955e4ec8eef9643e231184b2adde0b5e/src/layout.js#L192).

#### Analysis
`height` parameter can have [`fluid`](https://github.com/rtCamp/amphtml/blob/b2cd4622955e4ec8eef9643e231184b2adde0b5e/src/layout.js#L408) as value or any value from below list:
`width` can have [`auto`](https://github.com/rtCamp/amphtml/blob/b2cd4622955e4ec8eef9643e231184b2adde0b5e/src/layout.js#L399) as value or any value from below list:
- [Number](https://github.com/rtCamp/amphtml/blob/b2cd4622955e4ec8eef9643e231184b2adde0b5e/src/layout.js#L193)
  - Example: `height=100`, `width=200`
- [String containing Relative and OtherLengthUnits](https://github.com/rtCamp/amphtml/blob/b2cd4622955e4ec8eef9643e231184b2adde0b5e/src/layout.js#L199) (px|em|rem|vh|vw|vmin|vmax|cm|mm|q|in|pc|pt)
  - Example: `height="100px"`, `width="100vw"`
- [String containing Number](https://github.com/rtCamp/amphtml/blob/b2cd4622955e4ec8eef9643e231184b2adde0b5e/src/layout.js#L202)
  - Example: `height="100"`, `width="200"`
- Values other than above will be rejected ([width](https://github.com/rtCamp/amphtml/blob/b2cd4622955e4ec8eef9643e231184b2adde0b5e/src/layout.js#L400) value check, [height](https://github.com/rtCamp/amphtml/blob/b2cd4622955e4ec8eef9643e231184b2adde0b5e/src/layout.js#L409) value check)
  - Example: `height="100%"`, `width="100%"`, `height="auto"`, `width="fluid"`

> If input is number, it will be converted to `px` by default for both [number](https://github.com/rtCamp/amphtml/blob/b2cd4622955e4ec8eef9643e231184b2adde0b5e/src/layout.js#L194) and [stringNumber](https://github.com/rtCamp/amphtml/blob/b2cd4622955e4ec8eef9643e231184b2adde0b5e/src/layout.js#L203).

Thank you @pradeep910 and @ediamin to provide suggestion 🙂
I am updating as per the above research.

Reply:
Any suggestion for this error:
<img width="1264" alt="image" src="https://user-images.githubusercontent.com/75766877/121341414-43f1af80-c93e-11eb-9eed-9f5716432858.png">

This error only comes if there are more than one `amp-soundcloud` in [MediaQuery - Test](https://github.com/rtCamp/amphtml/blob/773e788c7ae429aa5fcefe459f0ed10b13bec9eb/extensions/amp-soundcloud/1.0/storybook/Basic.amp.js#L78).